src/sidebar/components/MarkdownView.tsx

```diff
@@ -35,6 +41,12 @@ export default function MarkdownView({
     });
   }, [markdown]);
 
+  useEffect(() => {
+    if (hasMentions) {
+      renderMentionTags(content.current!, mentions);
```

Comment:
I'm looking at the code for `renderMentionTags` and it is a bit confusing because it looks like it doesn't change anything at first glance. The `elementForMention` function does actually modify the element, but this isn't obvious from the name.

Reply:
Yes, exactly.

Reply:
Ok, since the function is not new from this PR, I'll add that to my todo list and address it separately, if that's ok.

Reply:
Here https://github.com/orgs/hypothesis/projects/153/views/1?pane=issue&itemId=97415193